src/Services/GithubApiService.ts

```diff
@@ -120,36 +99,25 @@ export class GithubApiService extends GithubRepository {
         CONSTANTS.DEFAULT_GITHUB_RETRY_DELAY,
       );
       const response = await retry.fetch<Promise<T>>(async ({ attempt }) => {
-        const res = await soxa.post("", {}, {
-          data: { query: query, variables },
-          headers: {
-            Authorization: `bearer ${TOKENS[attempt]}`,
-          },
-        });
-        if (res?.data?.errors) {
-          return this.handleError(res?.data?.errors);
-        }
-        return res;
-      }) as QueryDefaultResponse<{ user: T }>;
+        return await requestGithubData(
+          query,
+          variables,
+          TOKENS[attempt],
+        );
+      });
 
-      return response?.data?.data?.user ??
-        new ServiceError("not found", EServiceKindError.NOT_FOUND);
+      return response;
     } catch (error) {
-      if (error instanceof ServiceError) {
-        Logger.error(error);
-        return error;
+      if (error.cause instanceof ServiceError) {
+        Logger.error(error.cause.message);
+        return error.cause;
       }
-      // TODO: Move this to a logger instance later
       if (error instanceof Error && error.cause) {
         Logger.error(JSON.stringify(error.cause, null, 2));
       } else {
         Logger.error(error);
       }
-
-      return new ServiceError(
-        "Rate limit exceeded",
-        EServiceKindError.RATE_LIMIT,
-      );
+      return new ServiceError("not found", EServiceKindError.NOT_FOUND);
```

Comment:
I thought this line would be better to use "unknown error"  when none of the error conditions apply. How about that?

Reply:
I think this is another scope, maybe you can do it.

Reply:
I agree with you, this can be another class. But this PR is related to Retries. 

The class name, or helper is about another scope, i mean, another subject.
I think you can open another issue, so we can work only in this kind of improvements.